src/actions/clipboard.ts

```diff
@@ -80,43 +88,69 @@ export class Clipboard {
 
   /**
    * Create and register the keyboard shortcut for the cut action.
+   * Identical to the one in core but adds a toast after successful cut.
    */
   private registerCutShortcut() {
+    const oldCutShortcut =
+      ShortcutRegistry.registry.getRegistry()[ShortcutItems.names.CUT];
+    if (!oldCutShortcut)
+      throw new Error('No cut keyboard shortcut registered initially');
+
+    this.oldCutCallback = oldCutShortcut.callback;
+
     const cutShortcut: ShortcutRegistry.KeyboardShortcut = {
       name: Constants.SHORTCUT_NAMES.CUT,
-      preconditionFn: this.cutPrecondition.bind(this),
+      preconditionFn: oldCutShortcut.preconditionFn,
       callback: this.cutCallback.bind(this),
-      keyCodes: [
-        createSerializedKey(KeyCodes.X, [KeyCodes.CTRL]),
-        createSerializedKey(KeyCodes.X, [KeyCodes.ALT]),
-        createSerializedKey(KeyCodes.X, [KeyCodes.META]),
-      ],
-      allowCollision: true,
+      // The registry gives back keycodes as an object instead of an array
+      // See https://github.com/google/blockly/issues/9008
+      keyCodes: Object.values(oldCutShortcut.keyCodes ?? []),
+      allowCollision: false,
     };
 
+    ShortcutRegistry.registry.unregister(ShortcutItems.names.CUT);
     ShortcutRegistry.registry.register(cutShortcut);
   }
 
   /**
-   * Register the cut block action as a context menu item on blocks.
-   * This function mixes together the keyboard and context menu preconditions
-   * but only calls the keyboard callback.
+   * Register the cut block action as a context menu item.
+   * The context menu uses its own preconditionFn (that doesn't check
+   * if a gesture is in progress, because one always is in the context
+   * menu). It calls the cut callback that is shared between keyboard
+   * and context menu.
    */
   private registerCutContextMenuAction() {
     const cutAction: ContextMenuRegistry.RegistryItem = {
-      displayText: (scope) => `Cut (${getShortActionShortcut('cut')})`,
-      preconditionFn: (scope) => {
-        const ws = scope.block?.workspace;
-        if (!ws) return 'hidden';
+      displayText: (scope) =>
+        `Cut (${getShortActionShortcut(Constants.SHORTCUT_NAMES.CUT)})`,
 
-        return this.cutPrecondition(ws) ? 'enabled' : 'disabled';
+      // This is similar to the preconditionFn for the keyboard shortcut in core,
+      // but it doesn't have the check if a gesture is in progress.
+      preconditionFn: (scope) => {
+        const focused = scope.focusedNode;
+
+        if (!focused || !isCopyable(focused)) return 'hidden';
+
+        const workspace = focused.workspace;
+        if (
+          !workspace.isReadOnly() &&
+          isDeletable(focused) &&
+          focused.isDeletable() &&
+          isDraggable(focused) &&
+          focused.isMovable() &&
+          !focused.workspace.isFlyout
+        )
+          return 'enabled';
+
+        return 'disabled';
```

Comment:
Could this be factored out since it appears to be shared with copy?

Reply:
Done